src/main/java/com/suse/salt/netapi/utils/ClientUtils.java

```diff
@@ -83,4 +83,20 @@ public static String getModuleNameFromFunction(final String function)
         }
         return function.split("\\.")[0];
     }
+
+    /**
+     * Extract the function name from a given function(containing module and function name).
+     * In case of e.g. "test.ping", this method will return "ping".
+     *
+     * @param function string containing module and function name (e.g. "test.ping")
+     * @return the function name
+     * @throws IllegalArgumentException if a given function string does not contain a '.'
+     */
+    public static String getFunctionNameFromFunction(final String function)
+            throws IllegalArgumentException {
+        if (!function.contains(".")) {
+            throw new IllegalArgumentException(function);
+        }
+        return function.split("\\.")[1];
+    }
```

Comment:
I would like to suggest one more change to the patch: In the current version we have two helper methods that are mostly equal to each other, just one of them returns the first and the other returns the second array element. In order to simplify this a bit, couldn't we just have one method `splitFunction(final String function)` that would return the array as returned by `function.split("\\.")`, and then in the constructor assign module and function name like this?

```java
String[] splitFunction = splitFunction(function);
this.moduleName = splitFunction[0];
this.functionName = splitFunction[1];
```